registry/storage/purgeuploads.go

```diff
@@ -0,0 +1,144 @@
+package storage
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	storageDriver "github.com/docker/distribution/registry/storage/driver"
+)
+
+// TODO: logrus
+
+// uploadData defines the temporary files created during a layer upload
+type uploadData struct {
+	dataFilePath string
+	dateFilePath string
+	startedAt    time.Time
+}
+
+// TODO(richard): return containing folder
+func (ud *uploadData) Files() []string {
+	files := []string{ud.dataFilePath}
+	return append(files, ud.dateFilePath)
+}
+
+// PurgeUploads deletes files from the upload directory
+// created before olderThan.  The list of files deleted and errors
+// encountered are returned
+func PurgeUploads(driver storageDriver.StorageDriver, olderThan time.Time, actuallyDelete bool) ([]string, []error) {
+	// TODO(richard): normalize error type
+	uploadData, errors := getOutstandingUploads(driver)
+	var deleted []string
+	for _, uploadData := range uploadData {
+		for _, file := range uploadData.Files() {
+			if uploadData.startedAt.Before(olderThan) {
+				var err error
+				if actuallyDelete {
+					err = driver.Delete(file)
+				}
+				if err == nil {
+					deleted = append(deleted, file)
```

Comment:
Is the file actually deleted even if actuallyDelete is false? Seems we do nothing with the file